git/objects/commit.py

```diff
@@ -315,6 +316,53 @@ def stats(self) -> Stats:
             text = self.repo.git.diff(self.parents[0].hexsha, self.hexsha, '--', numstat=True)
         return Stats._list_from_string(self.repo, text)
 
+    @property
+    def trailers(self) -> Dict:
+        """Get the trailers of the message as dictionary
+
+        Git messages can contain trailer information that are similar to RFC 822
+        e-mail headers (see: https://git-scm.com/docs/git-interpret-trailers).
+
+        This funcions calls ``git interpret-trailers --parse`` onto the message
+        to extract the trailer information. The key value pairs are stripped of
+        leading and trailing whitespaces before they get saved into a dictionary.
+
+        Valid message with trailer:
+
+        .. code-block::
+
+            Subject line
+
+            some body information
+
+            another information
+
+            key1: value1
+            key2 :    value 2 with inner spaces
+
+        dictionary will look like this:
+        .. code-block::
+
+            {
+                "key1": "value1",
+                "key2": "value 2 with inner spaces"
+            }
+
+        :return: Dictionary containing whitespace stripped trailer information
+
+        """
+        d = {}
+        cmd = ['git', 'interpret-trailers', '--parse']
+        proc: Git.AutoInterrupt = self.repo.git.execute(cmd, as_process=True, istream=PIPE)  # type: ignore
```

Comment:
Can someone help with this?
I don't know how to get mypy to accept `subprocess.PIPE` as argument

Reply:
Maybe @Yobmod could help here.

Reply:
It looks like the isstream type annotation is too narrow in cmd.Git.execute().
It should be int | IO | None if we accept any args that Popen does.

But execute() needs a lot more annotations anyway to clear up the remaining str | bytes problems from py2.
So you can update the type annotation, ork leave the type: ignore is ok for now.
Once updated, mypy should then warn that the type: ignore is no longer needed.